auditlog_security/models/auditlog_rule.py

```diff
@@ -0,0 +1,127 @@
+            [('model_id.model', '=', model_name)], limit=1)
+        return rule.log_selected_fields_only
+
+    @api.model
+    def get_auditlog_fields(self, model):
+        res = super(AuditlogRule, self).get_auditlog_fields(model)
+        if self._get_log_selected_fields_only(model._name):
+            selected_field_names = self._get_field_names_of_rule(model._name)
+            # we re-use the checks on non-stored fields from super.
+            res = [x for x in selected_field_names if x in res]
+        return res
+
+    @api.multi
+    def write(self, values):
+        cache_invalidating_fields = [
+            "state",
+            "auditlog_line_access_rule_ids",
+            "log_selected_fields_only",
+        ]
+        if any([field in values.keys() for field in cache_invalidating_fields]):
+            # clear cache for all ormcache methods.
+            self.clear_caches()
+        return super(AuditlogRule, self).write(values)
+
+    @api.onchange("model_id")
+    def onchange_model_id(self):
+        # if model changes we must wipe out all field ids
+        self.auditlog_line_access_rule_ids.unlink()
+
+    @api.model
+    def _get_view_log_lines_action(self):
+        assert(self.env.context.get('active_model'))
+        assert(self.env.context.get('active_ids'))
+        model = self.env['ir.model'].sudo().search([
+            ('model', '=', self.env.context.get('active_model'))
+        ])
+        domain = [
+            ('model_id', '=', model.id),
+            ('res_id', 'in', self.env.context.get('active_ids')),
+        ]
+        return {
+            "name": _("View Log Lines"),
+            "res_model": "auditlog.log.line",
+            "view_mode": "tree,form",
+            "view_id": False,
+            "domain": domain,
+            "type": "ir.actions.act_window",
+        }
+
+    @api.multi
+    def _create_server_action(self):
+        self.ensure_one()
+        code = \
+            "action = env['auditlog.rule']._get_view_log_lines_action()"
+        server_action = self.env['ir.actions.server'].sudo().create({
+            'name': "View Log Lines",
+            'model_id': self.model_id.id,
+            'state': "code",
+            'code': code
+        })
+        self.write({
+            'server_action_id': server_action.id
+        })
+        return server_action
+
+    @api.multi
+    def subscribe(self):
+        for rule in self:
+            server_action = rule._create_server_action()
+            server_action.create_action()
+        res = super(AuditlogRule, self).subscribe()
+        for rule in self:
+            rule.auditlog_line_access_rule_ids.regenerate_rules()
+        # rule now will have "View Log" Action, make that visible only for admin
+        if res:
+            self.action_id.write({
+                'groups_id': [(6, 0, [self.env.ref('base.group_system').id])]
+            }) 
+        return res
+
+    @api.multi
+    def unsubscribe(self):
```

Comment:
@hbrunn we somehow need to restore this functionality. User can either subscribe to a rule (so they can see changelogs etc), or unsubscribe from a rule, and stop seeing anything related to that model. Do you have yet another clever way to achieve this easily?

Reply:
@ntsirintanis In the 11.0 version the subscribing / unsubscribing was not a per-user thing, it was just the enabling or disabling of that rule, and maybe not even that, but mainly the creation or removal of the "View Log" server action on the model. My idea behind it was to mimick the "Mass Edit" module in this, where you can say "Create server action".

So maybe it's just that the naming of the button is off and it should just be "Create Server action" and/or "Remove server action". Perhaps you can check how OCA mass edit (now called server_action_mass_edit) calls it, although that implementation changed also and you have to actually manually create a server action yourself in order to use mass editing. But perhaps you can look at an older major-Odoo-version-version of that module.